Timeline

V1T0R-CM opened the pull request.

GabrielCmrg committed:
feat: add signup component page

This page have only simple blocks for now, to work as placeholder for development.

GabrielCmrg committed:
chore: add signup component page to app

Use the router to render component on route /sign-up.

GabrielCmrg committed:
feat: add content on left side of the component.

Now signup has some content to style.

GabrielCmrg committed:
feat: change styles on signup page

GabrielCmrg committed:
feat: add a form to signup

GabrielCmrg committed:
feat: add styles to signup form

GabrielCmrg committed:
feat: add responsivity to styles.

GabrielCmrg committed:
feat: add responsivity to forms

GabrielCmrg committed:
feat: create controlled inputs

These controlled inputs are used to send an alert to fill all fields before submitting.
It is also possible to access individual input values this way.

GabrielCmrg committed:
feat: add API call and redirect or alerts the user

GabrielCmrg committed:
feat: disable button and inputs while waiting for API to respond.

GabrielCmrg committed:
Merge branch 'feature/sign-up' of github.com:GabrielCmrg/projeto17-linkr-front into feature/login

GabrielCmrg committed:
feat: add function to make a login request

This function uses the same structure as the other API function.
It returns more abrangent objects so the component can do what it needs.

V1T0R-CM committed:
componente Publication

V1T0R-CM committed:
correção nome das pastas

V1T0R-CM committed:
Merge branch 'dev' into timeline

GabrielCmrg committed:
feat: add login component as a page

This component has lots of features from signup component.

GabrielCmrg committed:
feat: create context to save token

The token is received from the localStorage.
Every time it changes the localStorage is updated.

GabrielCmrg committed:
feat: change user token on the context

Passes the returned token from the login route to the context for all the aplication.

GabrielCmrg committed:
chore: add .env variables

GabrielCmrg committed:
refactor: extract common lines as new components

There was some lines being replicated so i changed them to be components instead.
Also made the submit handler async so it wont need double function calls.

GabrielCmrg committed:
feat: change API call to be on the services

GabrielCmrg committed:
fix: change token to be received from context instead of localStorage

GabrielCmrg committed:
refactor: change style and use of context for image

GabrielCmrg reviewed (commented):
There are missing connections to the API.

V1T0R-CM committed:
Merge pull request #5 from GabrielCmrg/feature/login

Feature/login

GabrielCmrg committed:
feat: use of localStorage to get the image

The header and the prompt to make a post used to have a ulr placeholder.
This commit aims to get the image from the localstorage instead.

GabrielCmrg committed:
feat: populate the user image

Now when there is a login the user token and image are populated

GabrielCmrg committed:
fix: if statement and alt props

GabrielCmrg committed:
fix: hotfix on if statement

PauloGabrieel committed:
Merge pull request #6 from GabrielCmrg/feature/profile-photo

Feature/profile photo

PauloGabrieel committed:
Merge branch 'timeline' of https://github.com/GabrielCmrg/projeto17-linkr-front into timeline

PauloGabrieel commented:
> There are missing connections to the API.

I'm solving this

PauloGabrieel committed:
feat: create getAllPost function in services/api.js
